nixos/modules/services/printing/cupsd.nix

```diff
@@ -305,6 +321,16 @@ in
                 [ -L "$i" ] && rm "$i"
               done
             fi
+
+            # A special case for `printer.conf` for which we know cups will move the declarative
+            # version of the file out of the way on common operations (e.g: printing). So the
+            # current response to this annoying situation is to systematically replace the cups
+            # version with the declaratively specified one (that is, only when `null != printersConf`).
+            if [ -e "${rootdir}/etc/cups/printers.conf" ]; then
+              ln -s --backup=simple --suffix='.O' \
+                "${rootdir}/etc/cups/printers.conf" "/var/lib/cups/printers.conf"
+            fi
+
```

Comment:
Not exactly. The difference that matters is that for this particular file we now systematically / always do the linking even when the file exist (in which case we do a backup).